packages/components/src/spacer/types.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { SpaceInput } from '../ui/utils/space';
+
+export interface InnerProps {
+	/**
+	 * Adjusts all margins.
+	 */
+	margin?: SpaceInput;
+	/**
+	 * Adjusts top and bottom margins.
+	 */
+	marginY?: SpaceInput;
+	/**
+	 * Adjusts left and right margins.
+	 */
+	marginX?: SpaceInput;
+	/**
+	 * Adjusts top margins.
+	 */
+	marginTop?: SpaceInput;
+	/**
+	 * Adjusts bottom margins.
+	 *
+	 * @default 2
+	 */
+	marginBottom?: SpaceInput;
+	/**
+	 * Adjusts left margins.
+	 */
+	marginLeft?: SpaceInput;
+	/**
+	 * Adjusts right margins.
+	 */
+	marginRight?: SpaceInput;
+	/**
+	 * Adjusts all padding.
+	 */
+	padding?: SpaceInput;
+	/**
+	 * Adjusts top and bottom padding.
+	 */
+	paddingY?: SpaceInput;
+	/**
+	 * Adjusts left and right padding.
+	 */
+	paddingX?: SpaceInput;
+	/**
+	 * Adjusts top padding.
+	 */
+	paddingTop?: SpaceInput;
+	/**
+	 * Adjusts bottom padding.
+	 */
+	paddingBottom?: SpaceInput;
+	/**
+	 * Adjusts left padding.
+	 */
+	paddingLeft?: SpaceInput;
+	/**
+	 * Adjusts right padding.
+	 */
+	paddingRight?: SpaceInput;
+}
+
+export interface Props extends InnerProps {
+	children?: ReactNode;
+}
```

Comment:
I don't think we need the `InnerProps` separated out anymore. I believe I did that just to make it possible to have it without `children` on the styled component, but that's not necessary here. You might want to recombine them.

Reply:
Good point! Done in https://github.com/WordPress/gutenberg/pull/33555/commits/2a0cf2bcaae967a9e18338eb8cce3a305d2de88b